dietpi/func/dietpi-set_hardware

```diff
@@ -2259,6 +2259,96 @@ _EOF_
 
 	}
 
+	#/////////////////////////////////////////////////////////////////////////////////////
+	# Dedicated GPU
+	# INPUT_DEVICE_VALUE
+	# - Uninstall
+	# - Intel
+	# - Nvidia
+	# - AMD
+	#/////////////////////////////////////////////////////////////////////////////////////
+	DedicatedGPU_Main(){
+	
+		#Current GPU
+		local gpu_current=$(sed -n '/^[[:blank:]]*CONFIG_GPU_DRIVER=/{s/^[^=]*=//p;q}' /boot/dietpi.txt)
+		gpu_current=${gpu_current,,}
+		INPUT_DEVICE_VALUE=${INPUT_DEVICE_VALUE,,}
+		
+		#Steam installed for additional drivers?
+		local steam_installed=0
+		dpkg-query -s 'steam' &> /dev/null && steam_installed=1	
+		
+		if [[ $INPUT_DEVICE_VALUE == 'custom' ]]
+		then
+		
+			echo '0' &> /dev/null
+		
+		elif (( $G_HW_ARCH == 10 ))
+		then		
+		
+			if [[ $INPUT_DEVICE_VALUE == 'none' || 
+				$gpu_current != $INPUT_DEVICE_VALUE ]] #Change in HW vendor, uninstall 1st.
+			then
+
+				G_AGP mesa-vulkan-drivers:*
+
+				G_AGP i965-va-driver xserver-xorg-video-intel
+				G_AGP nvidia-driver nvidia-driver-libs:* libnvidia-glvkspirv:* nvidia-xconfig libgl1-nvidia-glx nvidia-settings nvidia-vulkan-common nvidia-vulkan-icd:*
+				G_AGP firmware-amd-graphics xserver-xorg-video-amdgpu
+				G_AGA
```

Comment:
To pass the wildcard through and avoid globbing, it needs to be quoted. `G_AGP` checks internally against the list of installed packages which one to purge, so there the asterisk should be expanded. And for performance reasons, I'd merge all purges into a single call.

And: `G_AGP` implies `G_AGA` in the meantime, so this does not need to be called afterwards 😉:
```suggestion
				G_AGP 'mesa-vulkan-drivers:*' i965-va-driver xserver-xorg-video-intel nvidia-driver 'nvidia-driver-libs:*' 'libnvidia-glvkspirv:*' nvidia-xconfig libgl1-nvidia-glx nvidia-settings nvidia-vulkan-common 'nvidia-vulkan-icd:*' firmware-amd-graphics xserver-xorg-video-amdgpu
```
Is there actually a chance that other package depend on any of these and might be unintentionally purged? We did not take care of this before, but to be failsafe we could also do:
```suggestion
				apt-mark auto $(dpkg --get-selections 'mesa-vulkan-drivers:*' i965-va-driver xserver-xorg-video-intel nvidia-driver 'nvidia-driver-libs:*' 'libnvidia-glvkspirv:*' nvidia-xconfig libgl1-nvidia-glx nvidia-settings nvidia-vulkan-common 'nvidia-vulkan-icd:*' firmware-amd-graphics xserver-xorg-video-amdgpu 2> /dev/null | mawk '{print $1}') 2> /dev/null && G_AGA
```

Reply:
> And for performance reasons, I'd merge all purges into a single call.

Yep, I wanted to keep the driver list per HW vendor on a dedicated line. Was easier to work with in code and see what package is for which device.

Maybe using your ```mark auto```, but adding? 
```
String='AMd packages'
String+=' Intel packages'
String+=' Nvidia Packages'
mark auto $String
```

Reply:
> Maybe using your `mark auto`, but adding?

Jep, that will also work, same as we do below for adding kernel headers with an array (best practice shell coding, so all entries are individually quoted):
```sh
array=('AMD1' 'AMD2')
array+=('Intel1' 'Intel2')
array+=('Nvidia1' 'Nvidia2')
apt-mark auto "${array[@]}"
```
Same can be applied to the other installs to always have a single `G_AGI` or better `G_AG_CHECK_INSTALL_PREREQ` call as well. The latter implies `G_AGUP` if required and does nothing if all packages are installed already.

Generally an alternative for are escaped line breaks:
```sh
apt-mark auto 1 2 3\
	4 5 6\
	7 8 9
```